src/Crypto/RNCryptor/V3/Decrypt.hs

```diff
@@ -102,10 +107,17 @@ decryptBlock :: RNCryptorContext
              -> ByteString
              -> (RNCryptorContext, ByteString)
 decryptBlock ctx cipherText = 
-  let clearText  = decryptCBC (ctxCipher ctx) (aesIV_ . rncIV . ctxHeader $ ctx) cipherText
-      !sz        = B.length cipherText
-      !newHeader = (ctxHeader ctx) { rncIV = (B.drop (sz - 16) cipherText) }
-      in (ctx { ctxHeader = newHeader }, clearText)
+  let clearText   = decrypt_ (ctxCipher ctx) (rncIV . ctxHeader $ ctx) cipherText
+      !newHMACCtx = update (ctxHMACCtx ctx) cipherText
+      !sz         = B.length cipherText
+      !newHeader  = (ctxHeader ctx) { rncIV = (B.drop (sz - 16) cipherText) }
+      in (ctx { ctxHeader = newHeader, ctxHMACCtx = newHMACCtx }, clearText)
+
+--------------------------------------------------------------------------------
+-- TBD:  need strictness here to overcome lazy pull from zip results stopping at
+-- first False?
+consistentTimeEqual :: ByteString -> ByteString -> Bool
+consistentTimeEqual a b = and $ B.zipWith (==) a b
```

Comment:
I'm confused! This function seems to merely zip the 2 `ByteString`s together `Word8` by `Word8` and then use `and` to assert equality. Why not use `==` in the first place? To be explicit:

 ``` haskell
consistentTimeEqual :: ByteString -> ByteString -> Bool
 consistentTimeEqual = (==)
```